plenoptic/simulate/models/frontend.py

```diff
@@ -1,96 +1,407 @@
+from typing import Tuple, Union, Callable
+
 import torch
 import torch.nn as nn
-from torchvision import transforms
-from ...tools.signal import make_disk
-import os
+import torch.nn.functional as F
+from torch import Tensor
+
+from ...tools.conv import same_padding
 from ...tools.display import imshow
+from ...tools.signal import make_disk
+from collections import OrderedDict
 
+__all__ = ["Gaussian", "CenterSurround", "LN", "LG", "LGG", "OnOff"]
```

Comment:
Are these models implementations of the ones Alex used? Then we should cite the proper paper (or his dissertation)

Reply:
Done in docstring up top.